lib/bitbucket_rest_api/request.rb

```diff
@@ -28,6 +28,20 @@ def delete_request(path, params={}, options={})
       request(:delete, path, params, options)
     end
 
+    def retry_token_refresh_errors
+      count = 0
+      begin
+        yield
+      rescue BitBucket::Error::RefreshToken
+        count += 1
+        if count < 3
```

Comment:
I think you want `<= 3` here if you want it retry 3 times after the first failed attempt.